### PR TITLE
Add configuration capability to desktop frontend

### PR DESCRIPTION
**Pull request by ohaz**

### CODE REVIEW

File: frontend-desktop/src/main.rs [hunk 1]
```diff
@@ -88,28 +92,46 @@ fn send(websocket: &Client, data: Value, event: &str) {
...
+}
+
+fn enable_buttons(ui: Weak<MainWindow>) {
+    ui.unwrap().invoke_set_buttons(true);
 }
 
 fn main() {
-    let config = Arc::new(read_config("./config"));
+
+    let config = read_config("./config");
```
Comment (t-animal) [inline frontend-desktop/src/main.rs]: Consider loading the file from the users config dir: https://docs.rs/dirs/1.0.4/dirs/fn.config_dir.html

File: frontend-desktop/src/main.rs [hunk 2]
```diff
@@ -145,24 +174,51 @@ fn main() {
...
+            }
+        }
+    });
+    ui.on_save_and_reconnect({
+        let main_window_weak = ui.as_weak();
+        let config = config.clone();
+        move |name, address| {
+            let new_config = CoffeeCallerConfig { username: name.as_str().into(), server: address.as_str().into() };
+            let toml = toml::to_string(&new_config).unwrap();
+            let file = OpenOptions::new().write(true).create(true).truncate(true).open("config.toml");
```
Comment (t-animal) [inline frontend-desktop/src/main.rs]: This file has as suffix, but in line 128 a different config is read? Is that on purpose?

**Review by t-animal — APPROVED** [review]
I could only quickly glance over it, but mostly lgtm. Is it maybe time to split the main.rs into modules (or whatever the correct term for rust ist)

File: frontend-desktop/src/main.rs (hunk 1 above)
Comment (ohaz) [reply frontend-desktop/src/main.rs]: Good idea, will do

File: frontend-desktop/src/main.rs (hunk 2 above)
Comment (ohaz) [reply frontend-desktop/src/main.rs]: The reading thing has some magic, I don't know how it works and I was super confused too, but it doesn't "need" a file type ending. It just chooses that automatically, no idea how. I'll check if I can add it to the reading part.

**Review by t-animal — APPROVED** [review]

**Pull request merged**